.travis.yml

```diff
@@ -36,66 +34,35 @@ jobs:
+        # ignores test_pythonpackage.py since it runs for too long
         - tox -- tests/ --ignore tests/test_pythonpackage.py
-        # (we ignore test_pythonpackage.py since these run way too long!!
-        #  test_pythonpackage_basic.py will still be run.)
       name: "Tox Pep8"
       env: TOXENV=pep8
-    - <<: *linting
+    - <<: *unittests
       name: "Tox Python 2"
       env: TOXENV=py27
-    - <<: *linting
+    - <<: *unittests
       name: "Tox Python 3 & Coverage"
       env: TOXENV=py3
       after_success:
         - coveralls
-
-    - &testing
-      stage: test
-      before_script:
-        # build docker image
-        - docker build --tag=p4a --file Dockerfile.py3 .
-        # Run a background process to make sure that travis will not kill our tests in
-        # case that the travis log doesn't produce any output for more than 10 minutes
-        - while sleep 540; do echo "==== Still running (travis, don't kill me) ===="; done &
-      script:
-        - >
-          docker run
-          -e CI
-          -e TRAVIS_JOB_ID
-          -e TRAVIS_BRANCH
-          -e ANDROID_SDK_HOME="/home/user/.android/android-sdk"
-          -e ANDROID_NDK_HOME="/home/user/.android/android-ndk"
-          p4a /bin/sh -c "$COMMAND"
-      after_script:
-        # kill the background process started before run docker
-        - kill %1
+    - &testapps
       name: Python 3 arm64-v8a
-      # overrides requirements to skip `peewee` pure python module, see:
-      # https://github.com/kivy/python-for-android/issues/1263#issuecomment-390421054
-      env:
-        COMMAND='. venv/bin/activate && cd testapps/ && python setup_testapp_python3_sqlite_openssl.py apk --sdk-dir $ANDROID_SDK_HOME --ndk-dir $ANDROID_NDK_HOME --requirements libffi,sdl2,pyjnius,kivy,python3,openssl,requests,sqlite3,setuptools --arch=arm64-v8a'
-    - <<: *testing
+      stage: build testapps
+      before_script: make docker/pull
+      script: make docker/run/make/testapps/python3/arm64-v8a
+    - <<: *testapps
       name: Python 3 armeabi-v7a
       os: osx
       osx_image: xcode11  # since xcode1.3, python3 is the default interpreter
       before_script:
         # installs java 1.8, android's SDK/NDK and p4a
         - make -f ci/makefiles/osx.mk
         - export JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
-        # Run a background process (like we do with linux tests)
-        - while sleep 540; do echo "==== Still running (travis, don't kill me) ===="; done &
-      script:
-        - >
-          cd testapps && python3 setup_testapp_python3_sqlite_openssl.py apk
-          --sdk-dir $HOME/.android/android-sdk
-          --ndk-dir $HOME/.android/android-ndk
-          --requirements libffi,sdl2,pyjnius,kivy,python3,openssl,requests,sqlite3,setuptools
-          --arch=armeabi-v7a
-    - <<: *testing
+      script: make testapps/python3/armeabi-v7a PYTHON_WITH_VERSION=python3
```

Comment:
Do we need to pass `PYTHON_WITH_VERSION=python3`?

We default to python3, so maybe we can remove it right?

Reply:
Good question, on osx I had to put it because the python3.6 binary itself didn't exist, making it fail when generating the virtualenv: `virtualenv --python=$(PYTHON_WITH_VERSION) $(VIRTUAL_ENV)`.
It's just than in osx they don't seem to symlink it